lib/components/src/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { Children, useCallback, useEffect, useRef, useState } from 'react';
+import { css, styled } from '@storybook/theming';
+import { AccordionContext } from './AccordionContext';
+
+import type { AddToMapFn, OpenMap } from './AccordionContext';
+
+type AccordionMap = Record<string, { index: number; id: string }>;
+
+type StateChange = {
+  id: string;
+  index: number;
+};
+
+export type AccordionProps = {
+  allowMultipleOpen?: boolean;
+  defaultOpen?: number | number[];
+  bordered?: boolean;
+  rounded?: boolean;
+  lined?: boolean;
+  onOpen?: (change: StateChange) => void;
+  onClose?: (change: StateChange) => void;
+} & React.HTMLAttributes<HTMLUListElement>;
+
+/**
+ *
+ * Controlling wrapper component that will control the open/close functionality to
+ * child ```<AccordionItem />```'s and style the other support components
+ * ```<AccordionHeader />``` and ```<AccordionBody />```
+ *
+ * Although each component can work independently and be controlled from the outside
+ * via an **open** prop, it is not recommended to use ```<AccordionItem />```,
```

Comment:
I think it's a good idea to use propNames like `isOpen` rather then `open`.

When passing props into `styled` props like `open` can make their way into the actual html, which can be unexpected.

Reply:
I agree. I might also change `narrow` to either `isCompact` or maybe even `size` with the values of `medium` and `small`. The latter is what is more common in the Storybook Design System.